http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -864,6 +869,16 @@ class DataprocInterpreter[F[_]: Parallel](
       // Set a SoftwareConfig property that makes the cluster have only one node
       Map("dataproc:dataproc.allow.zero.workers" -> "true")
     } else Map.empty[String, String]
+    // Reserve 3.5g to 4g (math.floor) memory for Welder, Jupyter, and Server processes. This value is arbitrary.
```

Comment:
Interesting.. so before was the driver allocating all the memory on the master VM, causing contention with Jupyter/welder containers?

Makes sense to reserve some memory for Juypter/welder/other processes. Wondering how you came up with 3.5G? (It seems reasonable, just curious)

Reply:
It seemed like this property wasn't being set at all? So I'm not entirely sure what the original value was.
I think the main issue was the driver wasn't allocating ENOUGH memory, it seemed like there's a `6g` constant in reference.conf.

Reply:
I calculated it by finding elsewhere in the code that Welder needed 0.5g, and saw online Jupyter also needed 0.5g MINIMUM. Then I figured 2 ish for system processes, so I rounded up slightly so in most cases it'd really be 4g left free (there's a case at 7.5 which will resolve to 4g for the JVM, which seemed like a good minimum.)

Reply:
OK - I seem to remember if `spark:spark.driver.memory` was not specified it would default to all the memory on the JVM. But not 100% sure. You should be able to check the `-Xmx` of the driver Java process and compare before/after.

Reply:
It seems like the spark.driver.memory will set the JVM memory to the same amount!

Reply:
The default value for `spark.driver.memory`, in Spark at least, [is `1g`](https://spark.apache.org/docs/latest/configuration.html). I do not know if Dataproc sets this intelligently. When users start clusters with `hailctl` we explicitly set the driver and executor memory (as well as memory overhead and a Hail-specific environment variable).

Regarding memory allocated to non-Spark driver-node processes: we default to 80% of the machine's RAM allocated to Spark. The rest is left open for Jupyter, etc.